pytorch_lightning/profiler/pytorch.py

```diff
@@ -263,32 +275,405 @@ def summary(self) -> str:
+
+    class PyTorchProfiler(LegacyPyTorchProfiler):  # noqa F811
+
+        START_ACTION = "on_fit_start"
+        STEP_FUNCTIONS = ("training_step_and_backward", "validation_step")
+
+        def __init__(
+            self,
+            output_filename: Optional[str] = None,
+            enabled: bool = True,
+            use_cpu: bool = True,
+            use_cuda: bool = True,
+            record_shapes: bool = True,
+            profile_memory: bool = True,
+            group_by_input_shapes: bool = False,
+            with_stack: bool = False,
+            export_to_chrome: bool = True,
+            path_to_export_trace: Optional[str] = None,
+            row_limit: int = 20,
+            sort_by_key: Optional[str] = None,
+            record_functions: Optional[List] = None,
+            local_rank: Optional[int] = None,
+            schedule: Optional[Callable] = torch.profiler.schedule(wait=1, warmup=1, active=2),
+            on_trace_ready: Optional[Callable] = None,
+            export_to_flame_graph: bool = True,
+            with_flops: bool = True,
+            metric: str = 'self_cpu_time_total',
+            record_module_names: bool = True,
+        ):
+            """
+            This profiler uses PyTorch's Autograd Profiler and lets you inspect the cost of
+            different operators inside your model - both on the CPU and GPU.
+            Available with PyTorch>=1.8, the profiler relies on the
+            `PyTorch Kineto project <https://github.com/pytorch/kineto>`__.
+
+            The PyTorch profiler API changed in 1.8, therefore this documentation will display both versions.
+
+            Args:
+                output_filename: optionally save profile results to file instead of printing
+                    to std out when training is finished. When using ``ddp``,
+                    each rank will stream the profiled operation to their own file
+                    with the extension ``_{rank}.txt``
+
+                enabled: Setting this to False makes this context manager a no-op.
+
+                use_cpu: Enables timing of CPU events.
+
+                use_cuda: Enables timing of CUDA events as well using the cudaEvent API.
+                    Adds approximately 4us of overhead to each tensor operation.
+
+                record_shapes: If shapes recording is set, information about input dimensions will be collected.
+
+                profile_memory: Whether to report memory usage
+
+                group_by_input_shapes: Include operator input shapes and group calls by shape.
+
+                with_stack: record source information (file and line number) for the ops (introduced in PyTorch 1.7)
+
+                export_to_chrome: Whether to export the sequence of profiled operators for Chrome.
+                    It will generate a ``.json`` file which can be read by Chrome.
+
+                path_to_export_trace: Directory path to export ``.json`` traces when using ``export_to_chrome=True``.
```

Comment:
If I understand the code correctly, this happens in on_trace_ready. We skip this when user defined on_trace_ready is provided. If my understanding is correct, we should update the documentation.